sklearn/utils/stats.py

```diff
@@ -1,18 +1,43 @@
 import numpy as np
 
 from .extmath import stable_cumsum
+from sklearn.utils.fixes import _take_along_axis
 
 
 def _weighted_percentile(array, sample_weight, percentile=50):
     """
     Compute the weighted ``percentile`` of ``array`` with ``sample_weight``.
+    If ``array`` is 2D, compute weighted ``percentile`` along axis=0.
     """
-    sorted_idx = np.argsort(array)
+    n_dim = array.ndim
+    if n_dim == 0:
+        return array[()]
+    # Remove trailing single-dimensional entries from the array shape
```

Comment:
I suspect that if the returned percentile needs to be a scalar for column vector input, this should be the responsibility of the caller, not of weighted percentile. I wouldn't expect this kind of automatic squeezing on other aggregation along an axis in Scientific Python.

Reply:
You're right. And I realise the problem was that `sample_weight` wasn't the same shape as `array`. fixed.